src/afl-fuzz.c

```diff
@@ -1288,13 +1304,39 @@ int main(int argc, char **argv_orig, char **envp) {
 
       /* afl-qemu-trace takes care of converting AFL_PRELOAD. */
 
+    } else if (afl->fsrv.frida_mode) {
+
+      afl_preload = getenv("AFL_PRELOAD");
+      u8 *frida_binary = find_binary("afl-frida-trace");
+      if (afl_preload) {
+
+        frida_afl_preload = alloc_printf("%s:%s", afl_preload, frida_binary);
+
+      } else {
+
+        frida_afl_preload = alloc_printf("%s", frida_binary);
+
+      }
+
+      ck_free(frida_binary);
+      OKF("Frida Mode setting LD_PRELOAD %s", frida_afl_preload);
+
+      setenv("LD_PRELOAD", frida_afl_preload, 1);
+      setenv("DYLD_INSERT_LIBRARIES", frida_afl_preload, 1);
+
     } else {
 
       setenv("LD_PRELOAD", getenv("AFL_PRELOAD"), 1);
       setenv("DYLD_INSERT_LIBRARIES", getenv("AFL_PRELOAD"), 1);
 
     }
 
+  } else if (afl->fsrv.frida_mode) {
+
+    OKF("Frida Mode setting LD_PRELOAD ./afl-frida-trace");
+    setenv("LD_PRELOAD", "./afl-frida-trace", 1);
+    setenv("DYLD_INSERT_LIBRARIES", "./afl-frida-trace", 1);
+
```

Comment:
this block needs to be updated similarly as well

Reply:
I hope this is what you meant? (e.g. using find_afl_binary rather than a fixed relative path)